lib/src/results.dart

```diff
@@ -211,23 +219,21 @@ class _RealmResultsIterator<T extends Object?> implements Iterator<T> {
   int _index;
   T? _current;
 
-  _RealmResultsIterator(RealmResults<T> results)
+  _RealmResultsIterator(RealmResults<T> results, {int index = 0})
```

Comment:
The is not used anywhere with the new parameter `index`. I suppose the return value of the `iterator` getter should looks like:
```dart
  @override
  Iterator<T> get iterator {
    ......
    return _RealmResultsIterator(results, _skipOffset);
```

Reply:
Hm, actually you already add `_skipOffset` when you call `realmCore.resultsGetObjectAt`. So, I'm not sure if we need this new `index` parameter of `_RealmResultsIterator` constructor.

Reply:
You are right, this is remnants from before I realised `skip` returns an iterable.